src/host/main.js

```diff
@@ -68,6 +68,23 @@ export async function hostMain({
 
     const callbackRequests = [];
 
+    const postCallbackWhenEnabled = async body => {
+      // ignore if not in callback mode
+      if (!callbackUrl) return;
+      const headers = {
+        'Content-Type': 'application/json',
+        ...(callbackHeader || {}),
+      };
+      await Promise.allSettled(callbackRequests);
```

Comment:
This function depends on the array that it modifies--it's a little unorthodox. If I understand the semantics (something I don't take for granted, which is what makes me suspicious of the pattern), the effective dependency is only on the final promise in the array at each invocation. In that case, it would be much more clear to track `latestRequest` and do something like

```js
latestRequest = latestRequest.catch(() => {}).then(fetch(/*...*/))
```

This is the kind of thing I had in mind when I suggested re-thinking `callbackRequests`. However, given [the redesign we're actively discussing](https://github.com/w3c/aria-at-app/issues/818#issuecomment-2035810525), I don't think it's worthwhile to spend too much time honing this implementation.